### DISCUSSION TITLE
Vcenter custom attributes pruning

### DISCUSSION THREAD

**Discussion by joshb89** [bb-Ricardo/netbox-sync#278] [General]
I wasn't able to find it in the documentation, but if you turn on custom attribute syncing for vmware, will this tool remove custom attribute fields in netbox if they get removed in vmware?

I tested it out this morning by syncing a new custom attribute to netbox and then changing the field of it in vmware and syncing one more time to netbox. What appears to happen is the original field is untouched in netbox and a new field is added as well. I couldn't find anything in the settings to be able to prune these fields in netbox.

**Comment by bb-Ricardo** [comment]
Hi,

I don't think this is currently implement.

So if you remove the custom field in VMware, it should be removed in NetBox as well. Is this correct?

**Reply by joshb89** [reply]
That is correct. What it is doing currently is if you either rename or remove a custom field in vmware, the custom field remains the same in netbox. I tested it by renaming the field in vmware and issuing a sync. The original custom field stayed on netbox and also created a new entry in netbox with the new name.

**Comment by bb-Ricardo** [comment]
Would you mind opening an issue for that. Then it's rasier to track and try to add this to one of the next versions.

Thank you.

**Comment by ngocnd1** [comment]
Hi bro, how could you use custom field to sync? I tried to sync but it isnt work

**Reply by bb-Ricardo** [reply]
Hi, your description of your setup is a bit vague. Could you please be a bit more precise of what exactly you trying to do and what is not working?

**Reply by ngocnd1** [reply]
firstly I using netbox-sync-83ceexxxxxxxx version and netbox version 3.4.5, both of them install directly not via docker
My code is
[common]
[netbox]
api_token = xxxxx
host_fqdn = x.x.x.x
port = 443
validate_tls_certs = false
[source/vcenter1]
type = vmware
host_fqdn = x.x.x.x
port = 443
sync_custom_attributes = false
vm_custom_object_attributes = Contact = Contact.Create Date = Create Date.Environment = Environment.Physical CPU Cores = Physical CPU Cores.Memory = Memory.Last Backup = Last Backup
track_vm_host = true
cluster_site_relation = .* = xxx
username = xxxx
password = xxxx
permitted_subnets = 172.0.0.0/8, 10.0.0.0/8

Eg. Contact(label on netbox) = Contact(Label on Vmware)
Even i use internal name vcsa_contact = Contact. It still isn't work

Is there any help, before i use this code via netbox and netbox-sync via docker it is still running well, Thanks Ricardo

**Reply by bb-Ricardo** [reply]
https://github.com/bb-Ricardo/netbox-sync/blob/3402b45e696f8573af9588f4a2f1d86b1dd3384f/settings-example.ini#L296-L304

you need to separate multiple attributes by `,` comma.

**Reply by ngocnd1** [reply]
i already tried

**Reply by ngocnd1** [reply]
> https://github.com/bb-Ricardo/netbox-sync/blob/3402b45e696f8573af9588f4a2f1d86b1dd3384f/settings-example.ini#L296-L304
> 
> you need to separate multiple attributes by `,` comma.

Could you give me an example what field in netbox we take to sync? Name internal or Lable
Example vm_custom_object_attributes = netbox1 = vcenter1, netbox2 = vcenter2,........

**Reply by bb-Ricardo** [reply]
> to get a list of available attributes use '-l DEBUG3' as cli param

the example in the config should work just fine
